RavenDB-11800 Don't allow adding node to the cluster when the leader initialize with port 0

aviv86 opened the pull request:
https://issues.hibernatingrhinos.com/issue/RavenDB-11800

aviv86 committed:
RavenDB-11800 : Don't allow adding node to the cluster when the leader initialize with port 0

aviv86 committed:
RavenDB-11800 : add test

ayende commented on src/Raven.Server/ServerWide/ServerStore.cs:
```diff
@@ -186,6 +186,7 @@ public long LastRaftCommitIndex
         private PoolOfThreads.LongRunningWork _clusterMaintenanceSetupTask;
         private PoolOfThreads.LongRunningWork _updateTopologyChangeNotification;
 
+        public bool ValidateNotRandomPort = true;
```
```suggestion
        public bool ValidateNonRandomPort = true;
```

ayende commented on src/Raven.Server/ServerWide/ServerStore.cs:
```diff
@@ -417,6 +418,18 @@ public ClusterTopology GetClusterTopology(TransactionOperationContext context)
+                    uri.Port == 0)
+                {
+                    throw new InvalidOperationException($"Failed to add node '{nodeUrl}' to cluster. " +
+                                                        "Adding nodes to cluster is forbidden when the leader has port '0' in 'Configuration.Core.ServerUrls' setting");
```
```suggestion
                                                        "Adding nodes to cluster is forbidden when the leader has port '0' in 'Configuration.Core.ServerUrls' setting. Define a fixed port for the node to enable cluster creation.");
```

ayende reviewed (commented):
Also, what about the other side, so we can't add a node that was started with port 0?

karmeli87 commented:
Not sure if you can tell that other side started with port 0..
What we can do is to reject requests to add us to a cluster, so the only valid cluster for starting with port 0 will be single node.

ayende commented:
We can give that indication in `RachisAdminHandler.GetNodeInfo`, and check that in `AddNode`

aviv86 committed:
RavenDB-11800
- fix PR comments
- validate fixed port for both the leader and the node to add
- add test case

ppekrol commented on src/Raven.Server/ServerWide/ServerStore.cs:
```diff
@@ -415,8 +417,20 @@ public ClusterTopology GetClusterTopology(TransactionOperationContext context)
         }
 
+        public bool HasFixedPort =>
+            Configuration.Core.ServerUrls == null ||
```
can this be `null`? thought that by default it is `localhost:8080`?
